metadata/metadata.go

```diff
@@ -1511,6 +1721,47 @@ func (serv *MetadataServer) GetModels(stream pb.Metadata_GetModelsServer) error
 	})
 }
 
+func (serv *MetadataServer) GetEquivalent(ctx context.Context, req *pb.ResourceVariant) (*pb.ResourceVariant, error) {
+	noEquivalentResponse := &pb.ResourceVariant{}
+
+	var currentResource ResourceVariant
+	var resourceType ResourceType
+
+	switch req.Resource.(type) {
+	case *pb.ResourceVariant_SourceVariant:
+		currentResource = &sourceVariantResource{req.GetSourceVariant()}
+		resourceType = SOURCE_VARIANT
+	case *pb.ResourceVariant_FeatureVariant:
+		currentResource = &featureVariantResource{req.GetFeatureVariant()}
+		resourceType = FEATURE_VARIANT
+	case *pb.ResourceVariant_LabelVariant:
+		currentResource = &labelVariantResource{req.GetLabelVariant()}
+		resourceType = LABEL_VARIANT
+	case *pb.ResourceVariant_TrainingSetVariant:
+		currentResource = &trainingSetVariantResource{req.GetTrainingSetVariant()}
+		resourceType = TRAINING_SET_VARIANT
+	default:
+		return nil, fmt.Errorf("unknown resource variant type: %v", req.Resource)
+	}
+
+	resources, err := serv.lookup.ListForType(resourceType)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, res := range resources {
+		other, ok := res.(ResourceVariant)
+		if !ok {
+			continue
+		}
+		if currentResource.IsEquivalent(other) {
```

Comment:
can we remove the above conditional statement and just do 
```if ok && currentResource.IsEquivalent(other) {```

Reply:
if should never be !ok though. Otherwise something we're doing is really wrong